keys/location.go

```diff
@@ -0,0 +1,29 @@
+package keys
+
+import "strings"
+
+type Location struct {
+	keyspace []string
+	joiner   string
+	name     string
+}
+
+func NewLocation(keyspace ...string, joiner, name string) *Location {
```

Comment:
I need some sort of abstraction like this, but I'm not sure if this is right one. Do you have a good suggestion?